packages/astro/src/vite-plugin-jsx/index.ts

```diff
@@ -104,6 +105,11 @@ interface AstroPluginJSXOptions {
 	logging: LogOptions;
 }
 
+// Format inspired by https://github.com/vitejs/vite/blob/main/packages/vite/src/node/constants.ts#L54
+const SPECIAL_QUERY_REGEX = new RegExp(
```

Comment:
I discovered our JSX runtime will try to process files with special flags like `?raw` and `?astroContent`. This regex should prevent that